src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Deserialize.cs

```diff
@@ -41,12 +41,12 @@ public static void ObjectReferenceLoop()
                 }
             }";
 
-            Employee angela = JsonSerializer.Deserialize<Employee>(json, s_deserializerOptionsPreserve);
+            Employee angela = await JsonSerializerWrapperForString.DeserializeWrapper<Employee>(json, s_deserializerOptionsPreserve);
```

Comment:
I think all these tests should be made to target both stream-based and string-based serialization, since reference preservation logic contains bespoke async code.

If anything, that's more feedback on the design of the base [SerializerTests](https://github.com/dotnet/runtime/blob/9030b78f19b363da4ffcaa00d8ae4929b010fd48/src/libraries/System.Text.Json/tests/Common/SerializerTests.cs#L8-L10) class. I don't think we should have separate entrypoints for "stream" and "string", since both can be abstracted behind a single serializer interface (effectively the `JsonSerializerWrapperForString` API). The different modalities can then be tested separately using `ReferenceHandlerTests_Sync` or `ReferenceHandlerTests_Async` derived classes that specify only one SUT.

[We already do this in a few other places](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/JsonSerializerWrapperForString.cs#L123) but we're far from being consistent.

Reply:
Agreed. Per offline discussion, we'll leave this PR to simply share the tests as-is with src-gen, then tackle this with https://github.com/dotnet/runtime/issues/66463.